src/hostname/hostnamed.c

```diff
@@ -408,6 +409,62 @@ static int property_get_chassis(
 
         return sd_bus_message_append(reply, "s", name);
 }
+static const char * const dmi_invalid_list[] = {
+        "To be filled by OEM",
+        "TBD by OEM",
+        "Default String",
+        "Board Manufacturer",
+        "Board Vendor Name",
+        "Board Product Name",
+        NULL,	/* terminator */
+};
+
+static int is_dmi_valid(const char *field)
+{
+        int i = 0;
+
+        while (dmi_invalid_list[i]) {
+                if (strstr(field, dmi_invalid_list[i]))
+                        return 0;
+                i++;
+        }
+
+        return 1;
+}
+
+static int property_get_model(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        char *is_lenovo, *model;
+        int r;
+
+        /* We usually want product_name, except for Lenovo. */
+        r = read_one_line_file("/sys/class/dmi/id/sys_vendor", &is_lenovo);
+        if (r < 0)
+                return -ENOENT;
```

Comment:
properties are not supposed to fail. at least not so easily. failing because of oom is fine, but just because some system doesn't have DMI this shouldn't be propagate to the client. Just return an empty string in that case, i.e. using it as indicator for "we have no clue"